llama_index/schema.py

```diff
@@ -43,12 +43,17 @@ def class_name(cls) -> str:
         """
 
     def __getstate__(self) -> Dict[str, Any]:
-        state = super().__getstate__()
+        # using super().__getstate__() would also include problematic private variables
+        state = self.dict()
+
         # Remove common unpicklable entries
         state["__dict__"].pop("tokenizer", None)
         state["__dict__"].pop("tokenizer_fn", None)
         return state
```

Comment:
@logan-markewich I think we should remove the `__dict__` in these `state["__dict__"]` as well. It relates to https://github.com/run-llama/llama_index/issues/8895. I have made a quick fix in https://github.com/run-llama/llama_index/pull/8902, please have a review. Sorry for the inconvenience :pray: